### ISSUE TITLE
Error in CMake install

### ISSUE THREAD

**Issue by jsansevero** [talgalili/installr#144]
**Describe the bug**
After running install.CMake() I receive a dialog box attempting to open a .org file and the installation then is unable to continue

**To Reproduce**
Code:
installr::install.CMake()

**Expected behavior**
Install CMake

**Additional context**
I am running this install an admin partition on a local machine.

**Comment by benkraj** [comment]
This doesn't help you, but I have the same issue. R 3.6.0 --- whether in R or Rstudio. Did you get it working?

**Comment by matiasandina** [comment]
Same error, looks like `install.CMake()` tries to match for a pattern that doesn't work anymore?

```
installr::install.CMake
function (URL = "https://cmake.org/download/", ...) 
{
    page_with_download_url <- URL
    page <- readLines(page_with_download_url, warn = FALSE)
    pat <- "/files/v[0-9.]+/cmake-[0-9.]+-win32-x86.msi"
    target_line <- grep(pat, page, value = TRUE)
    m <- regexpr(pat, target_line)
    URL <- regmatches(target_line, m)
    URL <- paste("https://cmake.org", URL, sep = "")[1]
    install.URL(URL, ...)
}
```
It seems that they updated the website and the pattern no longer works to get the files.

**Comment by talgalili** [comment]
Any chance you can send a pull request for the fix?

**Comment by matiasandina** [comment]
I think CMake shifted all their releases to Github (see https://github.com/Kitware/CMake/releases).
Mine is not a fix, it's a direct linl to. 
What's your feel for a more robust fix for  `installr::install.CMake()`? 
I feel it should do something along

```
installr::install.CMake(){

# go to https://github.com/Kitware/CMake/releases
# parse the highest v available (or specified by user)
# check user OS 
# construct the url
# give feedback to user about what URL it's trying to hit (in case something like this happens)
}
```

**Comment by matiasandina** [comment]
This gets me the `.tar.gz` on Ubuntu. I don't have other OS to make this cross compatible but I think it should be close enough. 

Probably another `switch` can give the way to `install.URL` in windows or just `download.file()` for the `.tar.gz` in other OS

```
function (repo =  "https://github.com/Kitware/CMake/releases", cmake_version = NULL)
{
  # enforce quality control for repo and cmake version
  if (!is.null(cmake_version) & !is.character(cmake_version)){
    stop("cmake_version must be NULL or a character with exact version number")
  }
  page <- readLines(repo, warn = FALSE)
  
  # get architecture and OS
  arch_os <- unlist(strsplit(R.version$system, ", "))
  ver <- "v[0-9.]+/"
  target_line <- grep(ver, page, value = TRUE)
  m <- regexpr(ver, target_line)
  # get available versions and sort decreasing
  available_versions <- sort(unique(regmatches(target_line, m)), decreasing = TRUE)
  # remove the "/"
  available_versions <- gsub(x= available_versions, pattern = "/", replacement = "")
  # default to latest or user input ?
  if (is.null(cmake_version)){
  message("Looking available versions for latest version")
  print(available_versions)
  target_version <- gsub(x = available_versions[1], pattern = "v", replacement = "")
  target_version <- gsub(x = target_version, pattern="/", replacement = "")
  } else {
    # must be exact match
    if (!(paste0("v", cmake_version) %in% available_versions)){
      message(paste0("Provided version `", cmake_version, "` not found on available versions. See below:"))
      # print with no v to avoid confusion
      print(gsub(x = available_versions, pattern = "v", replacement = ""))
      stop("provided version not available.")
    } 
    target_version <- cmake_version
  }
  
  file <- switch(arch_os[2], 
               "mingw32" = paste0("win-", arch_os[1], ".msi"),
               "linux-gnu" = paste0("Linux-", arch_os[1], ".tar.gz"),
               "Darwin" = paste0("Darwin-", arch_os[1], ".tar.gz")
  )
  
  file <- paste0("cmake-", target_version, "-", file)
  
  # links look like 
  # version/cmake-3.17.0-Darwin-x86_64.dmg
  URL <- paste(repo, "download", paste0("v", target_version), file, sep="/")
  print(URL)
  
  #install.URL(URL, ...)
}
```

**Comment by talgalili** [comment]
Sounds good, would you be willing to send a pull request?

**Comment by talgalili** [comment]
Could everyone please check the latest 0.23.0 version on github, and tell me if it works for you now?
Thanks.

**Comment by talgalili** [comment]
I'm assuming this problem is fixed in 0.23.0
